From 38168ab852672e1cd59bfb66c44f003d5b0683e1 Mon Sep 17 00:00:00 2001
From: Yoni Farhadian <yonif@marvell.com>
Date: Wed, 17 Jul 2013 08:52:41 +0300
Subject: [PATCH 0846/1825] PPv2: fix: WA for PHY address set

https://github.com/MISL-EBU-System-SW/misl-windriver.git linux-3.4.69-14t2-read
commit ef327b4694a626b8a495e940a4e0f70a0b0b2471

        Re-Init the PHY after applying PHY address, insead of using port's PHY AN enable register
        The reason is that writing to port's PHY AN register causes PHY to lose its link

Change-Id: I7c734d0b0e8f9af5bb919793faaa91f29a9b6fa7
Signed-off-by: Yoni Farhadian <yonif@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/2689
Reviewed-by: Eli Nidam <elini@marvell.com>
Tested-by: Eli Nidam <elini@marvell.com>
Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 .../arm/plat-armada/mv_hal/pp2/gmac/mvEthGmacApi.c |    4 ----
 1 files changed, 0 insertions(+), 4 deletions(-)

diff --git a/arch/arm/plat-armada/mv_hal/pp2/gmac/mvEthGmacApi.c b/arch/arm/plat-armada/mv_hal/pp2/gmac/mvEthGmacApi.c
index 00f5848..a34ba50 100644
--- a/arch/arm/plat-armada/mv_hal/pp2/gmac/mvEthGmacApi.c
+++ b/arch/arm/plat-armada/mv_hal/pp2/gmac/mvEthGmacApi.c
@@ -585,10 +585,6 @@ void mvEthPhyAddrSet(int port, int phyAddr)
 {
 	unsigned int regData;
 
-	regData = MV_REG_READ(ETH_PHY_AN_CFG0_REG);
-	regData |= ETH_PHY_AN_EN_MASK(port);
-	MV_REG_WRITE(ETH_PHY_AN_CFG0_REG, regData);
-
 	regData = MV_REG_READ(ETH_PHY_ADDR_REG);
 
 	regData &= ~ETH_PHY_ADDR_MASK(port);
-- 
1.7.5.4

